dali/python/nvidia/dali/pipeline.py

```diff
@@ -339,6 +346,9 @@ def _start_run(self):
             if not self._last_iter:
                 self.iter_setup()
                 self._batches_to_consume += 1
+            # Special case to prevent a deadlock if user didn't release the only buffer
+            if not self._exec_async and self._prefetch_queue_depth == 1:
+                self._release_outputs()
```

Comment:
What if it was released already. It is better to tell user that he should release outputs than do it silently. It may be surprising for the user.

Reply:
QueuePolicy::ReleaseIdxs is no-op if the outputs haven't been release

Reply:
But thats right, I am currently trying to add this mechanism in Executor itself

Reply:
But thats right, I am currently trying to add this mechanism in Executor itself

Reply:
Thing is that _release_output is a private function not documented, so the user cannot really do this right now